### PR TITLE
Implement getPendingIncidentsBatch for Elasticsearch

### PR DESCRIPTION
**Pull request by npepinpe**
## Description

This PR adds the implementation of `getPendingIncidentsBatch` for Elasticsearch.

## Related issues

related to #24930

### CODE REVIEW

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    final var sourceFilter =
+        new SourceFilter.Builder()
+            .includes(
+                PostImporterQueueTemplate.KEY,
+                PostImporterQueueTemplate.POSITION,
+                PostImporterQueueTemplate.INTENT)
+            .build();
+
+    return new SearchRequest.Builder()
+        .index(pendingUpdateAlias)
```
Comment (entangled90) [inline zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:75]: :question: is this the same as before?

**Review by entangled90 — APPROVED** [review]
:rocket:

File: zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java (hunk 1 above)
Comment (npepinpe) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:75]: Yes, you were supposed to start here, but I wasn't very clear :smile:

Comment (entangled90) [reply zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/incident/ElasticsearchIncidentUpdateRepository.java:75]: yes I realized it after reviewing this PR  :laughing:
I was referring if this code is new or is taken from a current implementation

**Pull request merged**